doc/man3/STORE_INFO.pod

```diff
@@ -0,0 +1,146 @@
+ X509 *STORE_INFO_get0_CERT(const STORE_INFO *store_info);
+ X509_CRL *STORE_INFO_get0_CRL(const STORE_INFO *store_info);
+
+ const char *STORE_INFO_type_string(int type);
+
+ void STORE_INFO_free(STORE_INFO *store_info);
+
+ STORE_INFO *STORE_INFO_new_NAME(char *name);
+ STORE_INFO *STORE_INFO_new_PARAMS(DSA *dsa_params);
+ STORE_INFO *STORE_INFO_new_PKEY(EVP_PKEY *pkey);
+ STORE_INFO *STORE_INFO_new_CERT(X509 *x509);
+ STORE_INFO *STORE_INFO_new_CRL(X509_CRL *crl);
+
+ STORE_INFO *STORE_INFO_new_ENDOFDATA(void);
+
+=head1 DESCRIPTION
+
+These functions are primarly useful for applications to retrieve
+supported objects from B<STORE_INFO> objects and for scheme specific
+loaders to create B<STORE_INFO> holders.
+
+=head2 Types
+
+B<STORE_INFO> is an opaque type that's just an intermediary holder for
+the objects that have been retrieved by STORE_load() and similar
+functions.
+Supported OpenSSL type object can be extracted using one of
+STORE_INFO_get0_TYPE(), after which the B<STORE_INFO> should be freed
+with STORE_INFO_free().
+See L</SUPPORTED OBJECTS> for more information on the types that are
+supported.
+
+=head2 Functions
+
+STORE_INFO_get_type() takes a B<STORE_INFO> and returns the STORE type
+number for the object inside.
+STORE_INFO_get_type_string() takes a STORE type number and returns a
+short string describing it.
+
+STORE_INFO_get0_NAME(), STORE_INFO_get0_PARAMS(),
+STORE_INFO_get0_CERT() and STORE_INFO_get0_CRL() all take a
+B<STORE_INFO> and return the held object of the appropriate OpenSSL
+type provided that's what's held.
+
+STORE_INFO_free() frees a B<STORE_INFO>.
+
+STORE_INFO_new_NAME(), STORE_INFO_new_PARAMS(), STORE_INFO_new_PKEY(),
+STORE_INFO_new_CERT() and STORE_INFO_new_CRL() create a B<STORE_INFO>
+object to hold the given input object.
+
+STORE_INFO_new_ENDOFDATA() create a B<STORE_INFO> object that holds no
+data, and is used exclusively used to tell the caller that there is no
+more data.
+
+=head1 SUPPORTED OBJECTS
+
+Currently supported object types are, in STORE object types:
+
+=over 4
+
+=item STORE_INFO_NAME
+
+A name is exactly that, a name.  It's like a name in a directory, but
+formatted as a complete URI.
+For example, the path in URI C<file:/foo/bar/> could include a file
+named C<cookie.pem>, and in that case, the returned B<STORE_INFO_NAME>
+object would have the URI C<file:/foo/bar/cookie.pem>, which can be
+used by the application to get the objects in that file.
+This can be applied to all schemes that can somehow support a listing
+of object URIs.
+
+For C<file:> URIs that are used without the explicit scheme, or paths
+given to L<STORE_open_file(3)>, the returned name will be the path of
+each object, so if C</foo/bar> was given and that path has the file
+C<cookie.pem>, the name C</foo/bar/cookie.pem> will be returned.
+
+=item STORE_INFO_PARAMS
+
+Key parameters.
+
+=item STORE_INFO_PKEY
+
+A private/public key of some sort.
+
+=item STORE_INFO_CERT
+
+A X.509 certificate.
```

Comment:
"a" and "an" are always based on how the next word is *pronounced* not on how it is spelt. So in this case ex-five-oh-nine starts with a vowel so "an" is used.